Kudu.Core/Jobs/BaseJobRunner.cs

```diff
@@ -72,17 +71,61 @@ protected string WorkingDirectory
 
         protected JobSettings JobSettings { get; set; }
 
-        private static int CalculateHashForJob(string jobBinariesPath)
+        internal static bool JobDirectoryHasChanged(
+            Dictionary<string, FileInfoBase> sourceDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> workingDirectoryFileMap, 
+            Dictionary<string, FileInfoBase> cachedSourceDirectoryFileMap)
         {
-            var updateDatesString = new StringBuilder();
-            DirectoryInfoBase jobBinariesDirectory = FileSystemHelpers.DirectoryInfoFromDirectoryName(jobBinariesPath);
+            // enumerate all source directory files, and compare against the files
+            // in the working directory (i.e. the cached directory)
+            FileInfoBase foundEntry = null;
+            foreach (var entry in sourceDirectoryFileMap)
+            {  
+                if (workingDirectoryFileMap.TryGetValue(entry.Key, out foundEntry))
+                {
+                    if (entry.Value.LastWriteTimeUtc > foundEntry.LastWriteTimeUtc)
```

Comment:
Done. Note that in doing this, I'm changing the behavior from the previous revisions. I'm now treating any modifications/deletions to source files in the working directory as changes that will cause a new copy. File additions in the working directory won't trigger a copy (to allow for the case where a job might be writing temp files). I think this makes sense when viewed as a "repair" operation to modified/missing source files. Note this is also how the existing prod code behaves (since it's computing hash codes over the entire directories).